LibationWinForms/LiberateDataGridViewImageButtonColumn.cs

```diff
@@ -22,12 +22,42 @@ protected override void Paint(Graphics graphics, Rectangle clipBounds, Rectangle
 
 			if (value is (LiberatedState liberatedState, PdfState pdfState))
 			{
-				(string mouseoverText, Bitmap buttonImage) = GridEntry.GetLiberateDisplay(liberatedState, pdfState);
+				(string mouseoverText, Bitmap buttonImage) = GetLiberateDisplay(liberatedState, pdfState);
 
 				DrawButtonImage(graphics, buttonImage, cellBounds);
 
 				ToolTipText = mouseoverText;
 			}
 		}
+
+		private static (string mouseoverText, Bitmap buttonImage) GetLiberateDisplay(LiberatedState liberatedStatus, PdfState pdfStatus)
+		{
+			(string libState, string image_lib) = liberatedStatus switch
+			{
+				LiberatedState.Liberated => ("Liberated", "green"),
+				LiberatedState.PartialDownload => ("File has been at least\r\npartially downloaded", "yellow"),
+				LiberatedState.NotDownloaded => ("Book NOT downloaded", "red"),
+				_ => throw new Exception("Unexpected liberation state")
+			};
+
+			(string pdfState, string image_pdf) = pdfStatus switch
+			{
+				PdfState.Downloaded => ("\r\nPDF downloaded", "_pdf_yes"),
+				PdfState.NotDownloaded => ("\r\nPDF NOT downloaded", "_pdf_no"),
+				PdfState.NoPdf => ("", ""),
+				_ => throw new Exception("Unexpected PDF state")
+			};
+
+			var mouseoverText = libState + pdfState;
+
+			if (liberatedStatus == LiberatedState.NotDownloaded ||
+				liberatedStatus == LiberatedState.PartialDownload ||
```

Comment:
Oh, right. PartialDownload isn't a *saved* state, but it's still a representable condition. I guess it should also be included in the same enum so that places like this can use it